tempto-core/src/main/java/com/teradata/tempto/s3/S3Client.java

```diff
@@ -0,0 +1,384 @@
+
+        // Initialize AWS SDK credentials
+        BasicAWSCredentials credentials = null;
+        try {
+            credentials = new BasicAWSCredentials(keyId, secret);
+        } catch (Exception e) {
+            throw new AmazonClientException("Cannot initialize credentials", e);
+        }
+
+        // Create S3 client
+        client = AmazonS3ClientBuilder.standard()
+                .withCredentials(new AWSStaticCredentialsProvider(credentials))
+                .build();
+    }
+
+    /**
+     * Method to get folder contents
+     *
+     * @param path  path to get contents
+     * @return      list of object names
+     */
+    public ArrayList<String> getFolderContents(String path)
+    {
+        return getObjects(path);
+    }
+
+    /**
+     * Method to get folder contents (including folders)
+     *
+     * @param path              path to get contents
+     * @param includeFolders    true/false to include folders or not
+     * @return
+     */
+    public ArrayList<String> getFolderContents(String path, boolean includeFolders)
+    {
+        ArrayList<String> objects = new ArrayList<String>();
+
+        // If no forward slash at end of path, add one
+        if (!path.substring(path.length() - 1).equals("/")) {
+            path += "/";
+        }
+
+        for (String object : getAllObjects(path)) {
+            // Remove input path from object name and determine the number of forward slashes left
+            String objectName = object.replace(path, "");
+            int slashCount = objectName.length() - objectName.replace("/", "").length();
+
+            // If there are no forward slashes or there is one forward slash and it is the last charcter
+            // (indicates that the folder is located in this folder) then include in the output
+            if (slashCount == 0 || (slashCount == 1 && objectName.substring(objectName.length() - 1).equals("/"))) {
+                objects.add(object);
+            }
+        }
+
+        return objects;
+    }
+
+    /**
+     * Method to return if file or folder already exists
+     *
+     * @param path  path of file or folder
+     * @return      true/false
+     */
+    public boolean exists(String path)
+    {
+        return client.doesObjectExist(bucketName, path);
+    }
+
+    /**
+     * Method to create folder in S3
+     *
+     * @param path  folder path to create
+     */
+    public void createFolder(String path)
+    {
+        // If no forward slash at end of path, add one
+        if (!path.substring(path.length() - 1).equals("/")) {
+            path += "/";
+        }
+
+        // If folder already exists, throw exception
+        if (exists(path)) {
+            throw new RuntimeException("Folder already exists - could not create folder: " + path);
+        }
+
+        // Create meta-data for your folder and set content-length to 0
+        ObjectMetadata metadata = new ObjectMetadata();
```

Comment:
The way S3 handles folders is that it is an object with zero length and a forward slash at the end.